addons/crm_phone_valid/models/crm_lead.py

```diff
@@ -14,53 +14,47 @@
     import phonenumbers
 
     class FormatPhoneMixin(object):
-        def _check_phone(self, phone):
-            if not phone:
-                return phone
+        def _check_contact_number(self, contact_number, country=None):
+            if not contact_number:
+                return contact_number
             company = self.env.user.company_id
             try:
-                country = self.country_id and self.country_id.code or company.country_id.code or False
-                nbr = phonenumbers.parse(phone, country)
-            except Exception, e:
-                raise UserError(_('Unable to parse the phone number!'))
+                country = self.country_id and self.country_id.code or company.country_id.code or country
+                nbr = phonenumbers.parse(contact_number, country)
+            except Exception:
+                raise UserError(_('Unable to parse the contact (phone,mobile,fax) number!'))
```

Comment:
`parse` is rather a technical word... "validate" would be more easy to understand